docs/index.rst

```diff
@@ -257,6 +257,35 @@ disabled individually.
 See the :class:`~urllib3.util.retry.Retry` definition for more details.
 
 
+Stream
+------
+
+You may also stream your response and get data as they come (e.g. when using
+``transfer-encoding: chunked``). In this case, method
+:func:`~urllib3.response.HTTPResponse.stream` will return generator.
+
+::
+
+    >>> import urllib3, datetime
```

Comment:
That's fair. It might make sense to add another section to the docs about handling different kinds of responses where we can go into deeper detail, but I feel that it's important to keep examples brief for the landing index page.